lib/hyacinth/adapters/external_identifier_adapter/datacite/ezid_response.rb

```diff
@@ -0,0 +1,52 @@
+class Hyacinth::Adapters::ExternalIdentifierAdapter::Datacite::EzidResponse
+  attr_reader :response, :parsed_body_hash
+
+  def initialize(response)
+    @response = response
+    @parsed_body_hash = parse_body
+  end
+
+  def success?
+    @response.body[0..6] == 'success'
+  end
+
+  def error?
+    @response.body[0..5] == 'error'
+  end
+
+  def doi
+    match = @response.body.match(/doi\S+/)
+    match[0]
+  end
+
+  def http_status_code
+    @response.code
+  end
+
+  def http_server_message
+    @response.msg
+  end
+
+  def body
+    @response.body
+  end
+
+  def parse_body
+    parsed_body_hash = {}
+    # First, split by '\n' to get individual lines
+    parsed_body = @response.body.split("\n")
+    # Then split each line in 2 along ':'. NOTE: using #map!
```

Comment:
Okay to delete this comment and the line below it?

Reply:
Sure (these are ported)